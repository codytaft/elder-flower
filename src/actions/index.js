export const setCurrentUser = user => ({
  type: 'SET_CURRENT_USER',
  user
});

export const createCarer = user => ({
  type: 'CREATE_CARER',
  user
});

export const createElder = user => ({
  type: 'CREATE_ELDER',
  user
});
