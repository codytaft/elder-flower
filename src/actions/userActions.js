export const setCurrentUser = user => ({
  type: 'SET_CURRENT_USER',
  user
});

export const createUser = user => ({
  type: 'CREATE_USER',
  user
});
