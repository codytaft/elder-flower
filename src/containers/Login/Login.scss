.login-form {
  display: flex;
  flex-direction: column;
  align-content: center;
  width: 25rem;
  margin: auto;
}

.login-input {
  text-decoration: none;
  font-size: 32px;
  margin: 2rem 0 2rem 0;
  border-radius: 3rem;
  padding-left: 1rem;
  border: 0.1rem solid rgb(78, 45, 29);
  color: #4e2d1d;
  height: 4rem;
  background-color: seashell;
}

.login-submit-button {
  margin: 2rem auto 2rem auto;
  text-decoration: none;
  border: 0.2rem solid rgb(78, 45, 29);
  color: #4e2d1d;
  border-radius: 5rem;
  cursor: pointer;
  width: 25rem;
  height: 4rem;
  font-size: 32px;
  background-color: seashell;
}

.login-submit-button:hover {
  background-color: rgba(215, 184, 141, 1);
}
