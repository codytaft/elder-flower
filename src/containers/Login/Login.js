import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { setCurrentUser } from '../../actions';
import { getUser } from '../../helpers/fetchCalls';
import './Login.css';

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    const { history } = this.props;
    getUser(this.state.email);
    const location = { pathname: './dashboard' };
    history.push(location);
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit} className="login-form">
        <input
          onChange={this.handleChange}
          type="text"
          className="login-email-input login-input"
          value={this.state.email}
          name="email"
          placeholder="Email Address"
        />
        <input
          onChange={this.handleChange}
          type="text"
          className="login-password-input login-input"
          value={this.state.password}
          name="password"
          placeholder="Password"
        />
        <button className="login-submit-button">Submit</button>
      </form>
    );
  }
}

Login.propTypes = {
  setCurrentUser: PropTypes.func.isRequired,
  currentUser: PropTypes.object,
  history: PropTypes.object
};

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Login)
);
