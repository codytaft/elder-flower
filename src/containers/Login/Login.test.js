import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';
import { setCurrentUser } from '../../actions';
import { mapStateToProps, mapDispatchToProps } from './Login';
import bcrypt from 'bcryptjs';

jest.mock('bcryptjs', () => ({
  genSaltSync: () => 'sdkfj',
  hashSync: () => 'yournewhash',
  compareSync: () => true
}));

describe('Login', () => {
  let wrapper;
  it('should match the snapshot', () => {
    wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on change', () => {
    wrapper = shallow(<Login />);

    const mockEmailEvent = {
      target: { name: 'email', value: 'cody.taft@gmail.com' }
    };
    const mockPasswordEvent = {
      target: { name: 'password', value: '123' }
    };

    wrapper.instance().handleChange(mockEmailEvent);
    expect(wrapper.state('email')).toEqual(mockEmailEvent.target.value);
    wrapper.instance().handleChange(mockPasswordEvent);
    expect(wrapper.state('password')).toEqual(mockPasswordEvent.target.value);
  });

  it.skip('should get user on submit', async () => {
    let mockEvent = { preventDefault: () => jest.fn() };
    let mockUser = {
      email: 'cody.taft@gmail.com',
      password: '123'
    };
    let history = {
      length: 27,
      action: 'PUSH',
      location: { pathname: '/dashboard', search: '', hash: '', key: 'kzlimn' }
    };
    const email = 'cody.taft@gmail.com';
    const password = '123';
    await wrapper.setState({ email, password });
    await wrapper.instance().handleSubmit(mockEvent);
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockUser)
      })
    );

    wrapper = shallow(<Login />);
  });
});

describe('mapStateToProps', () => {
  it('should return an object with a currentUser object', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockState = {
      currentUser: { ...mockCurrentUser }
    };

    const expected = {
      currentUser: {
        id: 75,
        firstName: 'Cody',
        lastName: 'Taft',
        phoneNumber: '+19038511575',
        email: 'cody.taft@gmail.com',
        password: '123',
        contactName: 'Gaynell',
        contactPhone: '+17203304593',
        isElder: false
      }
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should invoke dispatch when using a function from mapDispatchToProps', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockDispatch = jest.fn();
    const actionToDispatch = setCurrentUser(mockCurrentUser);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setCurrentUser(mockCurrentUser);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
