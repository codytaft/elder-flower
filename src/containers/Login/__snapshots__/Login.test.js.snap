// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="login-email-input login-input"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          type="text"
          value=""
/>,
        <input
          className="login-password-input login-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="text"
          value=""
/>,
        <button
          className="login-submit-button"
>
          Submit
</button>,
      ],
      "className": "login-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-email-input login-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email Address",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-password-input login-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "login-submit-button",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="login-email-input login-input"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="text"
            value=""
/>,
          <input
            className="login-password-input login-input"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="text"
            value=""
/>,
          <button
            className="login-submit-button"
>
            Submit
</button>,
        ],
        "className": "login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-email-input login-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-password-input login-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "login-submit-button",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
